internal/pkg/filesystem/aferofs/factory.go

```diff
@@ -8,25 +8,55 @@ import (
 	"github.com/keboola/keboola-as-code/internal/pkg/filesystem/aferofs/localfs"
 	"github.com/keboola/keboola-as-code/internal/pkg/filesystem/aferofs/memoryfs"
 	"github.com/keboola/keboola-as-code/internal/pkg/filesystem/aferofs/mountfs"
-	"github.com/keboola/keboola-as-code/internal/pkg/log"
 )
 
-func NewLocalFs(logger log.Logger, rootDir string, workingDirRel string) (fs filesystem.Fs, err error) {
+func NewLocalFs(rootDir string, opts ...filesystem.Option) (fs filesystem.Fs, err error) {
 	backendFs, err := localfs.New(rootDir)
 	if err != nil {
 		return nil, err
 	}
-	return New(logger, backendFs, workingDirRel), nil
+	return New(backendFs, opts...), nil
 }
 
-func NewMemoryFs(logger log.Logger, workingDir string) (fs filesystem.Fs, err error) {
-	return New(logger, memoryfs.New(), workingDir), nil
+func NewMemoryFs(opts ...filesystem.Option) filesystem.Fs {
```

Comment:
`NewMemoryFs` nevracia uz `err`, kedze tam ziadna chyba nastat nemoze.
Predtym tam bol `err` aby to splnano interface `fsFactory`.
Na to som vytvoril samostatnu funkciu `NewMemoryFsOrErr`.